web/src/components/CreatePreAuthorizationModal.tsx

```diff
@@ -149,7 +150,7 @@ const CreatePreAuthorizationModal: React.FC<CreatePreAuthorizationModalProps> =
 
             try {
                 let authorizedAmount = new Decimal(authorizedAmountStr);
-                const decimals = token!.decimals;
+                const decimals = token?.decimals ?? mint?.decimals ?? 0;
```

Comment:
For decimals, shouldn't mint get priority in this since its native to the mint state?